core/src/main/resources/error/error-classes.json

```diff
@@ -54,6 +54,9 @@
   "IF_PARTITION_NOT_EXISTS_UNSUPPORTED" : {
     "message" : [ "Cannot write, IF NOT EXISTS is not supported for table: %s" ]
   },
+  "ILLEGAL_FORMAT_ARGUMENT_INDEX" : {
```

Comment:
Shall we have a unified error class for illegal format string? cc @karenfeng

Reply:
There are a couple different ways for a string to have an illegal format, right? I'm not sure how to unify them all, but we can definitely have a more generic class here to represent that a string cannot contain a given substring, eg:

```
"ILLEGAL_SUBSTRING" : {
   "message" : [ "%s cannot contain %s." ]
}
```

Reply:
@karenfeng bf5790d fix this

Reply:
@cloud-fan, does this LGTY? I don't have any more concerns.

Reply:
Gentle ping @cloud-fan @dongjoon-hyun